lib/graphql/object_type.rb

```diff
@@ -42,4 +42,26 @@ def interfaces=(new_interfaces)
   def kind
     GraphQL::TypeKinds::OBJECT
   end
+
+  # @return [GraphQL::Field] The field definition for `field_name` (may be inherited from interfaces)
+  def get_field(field_name)
+    fields[field_name] || get_interface_field(field_name)
+  end
+
+  # @return [Array<GraphQL::Field>] All fields, including ones inherited from interfaces
+  def all_fields
+    interface_fields = interfaces.inject({}) do |memo, iface|
+      memo.merge(iface.fields)
+    end
+    interface_fields.merge(self.fields).values
+  end
+
+  private
+
+  # Find the _last_ definition for `field_name`
+  def get_interface_field(field_name)
+    interfaces.reduce(nil) do |memo, iface|
```

Comment:
Hmm right! I guess what I'm saying is why not have `interface_fields` as a method itself, in which you can use `interfaces.reduce({}, :merge)` then you can replace that part in both `all_fields` and `get_interface_field` which becomes `interface_fields[field_name]`

Just a nit though, don't want to slow you down!